runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -343,11 +345,10 @@ public void clear(K key, W window) {
         }
 
         private void prepareStateBackend(K key) {
-          // Key for state request is shipped already encoded as ByteString,
-          // this is mostly a wrapping with ByteBuffer. We still follow the
-          // usual key encoding procedure.
-          // final ByteBuffer encodedKey = FlinkKeyUtils.encodeKey(key, keyCoder);
-          final ByteBuffer encodedKey = ByteBuffer.wrap(key.toByteArray());
+          // Key for state request is shipped encoded. It may be encoded with NESTED context which
```

Comment:
Don't we need to know for sure it was encoded as NESTED to perform such re-encoding? FWIW the test fails because of the attempt to decode a non-nested (not length prefixed key). ` [107, 49]`